### PR TITLE
04 - MVC Models

### PR DESCRIPTION
**Pull request by mar7in91Wolox**
# Summary
- I added the following [files/changes](https://gist.github.com/gzamudio/4aee3f1142d19b1eeb9ffa71442e1bb4).

# Trello Card
https://trello.com/c/D7KphCU4

# Screenshots
image

### CODE REVIEW

File: training-net/src/Models/Database/Movie.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace MvcMovie.Models
+{
+    public class Movie
+    {
+        public int ID { get; set; }
+        public string Title { get; set; }
+        public DateTime ReleaseDate { get; set; }
+        public string Genre { get; set; }
+        public decimal Price { get; set; }
```
Comment (truccomarcos) [inline training-net/src/Models/Database/Movie.cs]: I think you should use a `float` instead of a `decimal`

File: training-net/src/Startup.cs [hunk 2]
```diff
@@ -7,8 +7,11 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.HttpsPolicy;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using MvcMovie.Repositories.Database;
+
```
Comment (truccomarcos) [inline training-net/src/Startup.cs]: There shouldn't be more than one empty line separating the imports from the namespace definition

File: training-net/test/training-net-test.csproj [hunk 3]
```diff
@@ -1,16 +1,13 @@
...
+    <RootNamespace>training_net_test</RootNamespace>
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0"/>
+    <PackageReference Include="xunit" Version="2.3.1"/>
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.3.1"/>
+    <PackageReference Include="LocalizationCultureCore" Version="1.1.2"/>
+  </ItemGroup>
+</Project>
```
Comment (truccomarcos) [inline training-net/test/training-net-test.csproj]: Add an empty line at EOF :)

File: training-net/src/Models/Database/Movie.cs (hunk 1 above)
Comment (mar7in91Wolox) [reply training-net/src/Models/Database/Movie.cs]: Done!

File: training-net/src/Startup.cs (hunk 2 above)
Comment (mar7in91Wolox) [reply training-net/src/Startup.cs]: Done!

**Comment by mar7in91Wolox** [comment]
"Why are the migration files here? Shouldn't those be added in the next PR?"

@truccomarcos 
 In the [asociated Trello Card](https://trello.com/c/D7KphCU4) there is a link to this [Microsoft tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-mvc-app-xplat/adding-model?view=aspnetcore-2.1) that explains how to generate a "DataBase Model". I remember you teached me how to make a first migration although I were in MVC-Models branch.

**Comment by truccomarcos** [comment]
@mar7in91Wolox 
> In the asociated Trello Card there is a link to this Microsoft tutorial that explains how to generate a "DataBase Model". I remember you teached me how to make a first migration although I were in MVC-Models branch.

My bad :sweat_smile: Thanks for fixing that.

File: training-net/src/Models/Database/Movie.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+using System;
+
+namespace MvcMovie.Models
```
Comment (truccomarcos) [inline training-net/src/Models/Database/Movie.cs]: Update the namespace here

File: training-net/src/Repositories/Database/DataBaseContext.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#region Using
+using Microsoft.EntityFrameworkCore;
+using MvcMovie.Models;
+#endregion
+
+namespace MvcMovie.Repositories.Database
```
Comment (truccomarcos) [inline training-net/src/Repositories/Database/DataBaseContext.cs]: Here too

File: training-net/src/Repositories/Database/DataBaseContext.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+#region Using
```
Comment (truccomarcos) [inline training-net/src/Repositories/Database/DataBaseContext.cs]: I didn't notice earlier. Same as the migrations, this file is only needed for the database, so it should be on the next PR. We can leave it here though :smile:

File: training-net/src/Startup.cs [hunk 7]
```diff
@@ -7,8 +7,10 @@
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.HttpsPolicy;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
```
Comment (truccomarcos) [inline training-net/src/Startup.cs]: Same as with the migrations, the changes on this file are not needed until the database is used. We can leave them here though :smile:

Comment (mar7in91Wolox) [reply training-net/src/Startup.cs]: :)

**Pull request merged**